changelog.d/9548.removal

```diff
@@ -0,0 +1 @@
+/register now expects a `body.type` value of `m.login.appservice` when registering appservice users, to conform to the spec.
```

Comment:
```suggestion
Make `/_matrix/client/r0/register` expect a login type of `m.login.application_service` when an Application Service registers a user, to align with [the relevant spec](https://spec.matrix.org/unstable/application-service-api/#server-admin-style-permissions).
```

May be a bit wordy, but hopefully clearer. Also `s/m.login.appservice/m.login.application_service`.